pkg/routeagent/controllers/route/route.go

```diff
@@ -310,6 +326,43 @@ func (r *Controller) updateIptableRulesForInterclusterTraffic(inputCidrBlocks []
 	}
 }
 
+func (r *Controller) updateRoutingRulesForHostNetworkSupport(inputCidrBlocks []string, operation Operation) {
+	if operation == FlushRouteTable {
+		r.routeCacheGWNode.DeleteAll()
+		cmd := exec.Command("/sbin/ip", "r", "flush", "table", strconv.Itoa(RouteAgentHostNetworkTableID))
+		if err := cmd.Run(); err != nil {
+			// We can safely ignore this error, as this table
+			// won't exist in most nodes (only gateway nodes)
+			return
+		}
+	} else {
+		// These routing rules are required ONLY on the Gateway Node.
+		// On the non-Gateway nodes, we use iptable rules to support this use-case.
+		if r.isGatewayNode && r.cniIface != nil {
+			switch operation {
+			case AddRoute:
+				for _, inputCidrBlock := range inputCidrBlocks {
+					if !r.routeCacheGWNode.Contains(inputCidrBlock) {
+						r.routeCacheGWNode.Add(inputCidrBlock)
+						if err := r.configureRoute(inputCidrBlock, operation); err != nil {
+							klog.Errorf("Failed to add route for HostNetwork support on the Gateway node. %v", err)
```

Comment:
```suggestion
							klog.Errorf("Failed to add route %q for HostNetwork support on the Gateway node: %v", inputCidrBlock , err)
```

Perhaps it will help troubleshooting to output the route it was trying to add.

Reply:
thats a good point.